limitador/src/limit.rs

```diff
@@ -166,13 +330,283 @@ impl PartialEq for Limit {
+    pub enum TokenType {
+        // Predicates
+        EqualEqual,
+
+        //Literals
+        Identifier,
+        String,
+        Number,
+    }
+
+    #[derive(Eq, PartialEq, Debug)]
+    pub enum Literal {
+        Identifier(String),
+        String(String),
+        Number(i64),
+    }
+
+    #[derive(Eq, PartialEq, Debug)]
+    pub struct Token {
+        pub token_type: TokenType,
+        pub literal: Option<Literal>,
+        pub pos: usize,
+    }
+
+    pub struct Scanner {
+        input: Vec<char>,
+        pos: usize,
+    }
+
+    impl Scanner {
+        pub fn scan(condition: String) -> Result<Vec<Token>, usize> {
+            let mut tokens: Vec<Token> = Vec::with_capacity(3);
+            let mut scanner = Scanner {
+                input: condition.chars().collect(),
+                pos: 0,
+            };
+            while !scanner.done() {
+                match scanner.next_token() {
+                    Ok(token) => {
+                        if let Some(token) = token {
+                            tokens.push(token)
+                        }
+                    }
+                    Err(err) => {
+                        return Err(err);
+                    }
+                }
+            }
+            Ok(tokens)
+        }
+
+        fn next_token(&mut self) -> Result<Option<Token>, usize> {
+            let character = self.advance();
+            match character {
+                '=' => {
+                    if self.next_matches('=') {
+                        Ok(Some(Token {
+                            token_type: TokenType::EqualEqual,
+                            literal: None,
+                            pos: self.pos - 1,
+                        }))
+                    } else {
+                        Err(self.pos)
+                    }
+                }
+                '"' | '\'' => self.scan_string(character).map(Some),
+                ' ' | '\n' | '\r' | '\t' => Ok(None),
+                _ => {
+                    if character.is_alphabetic() {
+                        self.scan_identifier().map(Some)
+                    } else if character.is_numeric() {
+                        self.scan_number().map(Some)
+                    } else {
+                        Err(self.pos)
+                    }
+                }
+            }
+        }
+
+        fn scan_identifier(&mut self) -> Result<Token, usize> {
+            let start = self.pos;
+            while !self.done() && self.valid_id_char() {
+                self.advance();
+            }
+            Ok(Token {
+                token_type: TokenType::Identifier,
+                literal: Some(Literal::Identifier(
+                    self.input[start - 1..self.pos].iter().collect(),
+                )),
+                pos: start,
+            })
+        }
+
+        fn valid_id_char(&mut self) -> bool {
+            let char = self.input[self.pos];
+            char.is_alphanumeric() || char == '.' || char == '_'
```

Comment:
The dot (`.`) being part of this here makes me sad… I'd expect `foo.bar` to not simply be an identifier…

Reply:
Actually, it can be any utf-8 encoded string. Same for the `operand`. This code only applies to the `identifier` which is related to the descriptor entry key.